docs/cloud/connect/serverless/cloud-run.md

```diff
@@ -0,0 +1,180 @@
+        return
+    }
+    fmt.Printf("Connected to Dragonfly: %s\n", pong)
+
+    // Perform some test operations
+    err = client.Set(ctx, "test_key", name, 0).Err()
+    if err != nil {
+        fmt.Printf("Error setting key: %v\n", err)
+        return
+    }
+
+    value, err := client.Get(ctx, "test_key").Result()
+    if err != nil {
+        fmt.Printf("Error getting key: %v\n", err)
+        return
+    }
+    fmt.Printf("Retrieved value: %s\n", value)
+
+    // Close the connection
+    err = client.Close()
+    if err != nil {
+        fmt.Printf("Error closing connection: %v\n", err)
+    }
+}
+```
+
+The code uses two environment variables `DFADDR` and `DFPASS`.
+
+## Steps to Connect from the Cloud Run Function to a public data store
+
+### 1. Create a New Cloud Run Function
+
+1. Go to the [Cloud Run](https://console.cloud.google.com/run).
+2. Click **Write a function**.
+3. Choose **Go Runtime**.
+4. Provide a name for your function (e.g., `DragonflyConnector`).
+5. Expand the Containers section.
+6. Edit the **Container Port** to match with the data store's port.
+7. Add `DFADDR` and `DFPASS` environment variables.
+8. Click **Create**.
+
+### 2. Test the Cloud Run Function
+
+1. Click the **Test** button in the console.
+2. Create a new test event (you can use the default template).
+3. Run the test in cloud shell.
+4. Check the logs in **Logs** to verify the connection and the key-value pair operation.
+
+---
+
+## Connect to a Private Datastore
+
+Private datastores are hosted within a Virtual Private Cloud (VPC), which provides
+an isolated network environment. To enable your Cloud Run function to securely
+connect to a private Dragonfly datastore, follow these beginner-friendly steps:
+
+### 1. Set Up VPC Peering
+
+1. Create a VPC in your GCP account within the same region as your datastore.
+2. Establish a peering connection between your VPC and the datastore's VPC. This allows the two networks to communicate. For detailed guidance, refer to the [VPC Peering Connections documentation](../../connections.md).
+
+### 2. Adjust Firewall Rules
+
+1. Open the [VPC Network Console](https://console.cloud.google.com/networking/networks/list).
+2. Select your network and open the Firewall settings.
+3. Add an ingress rule to allow traffic from your datastore vpc. Put datastore vpc CIDR range in the Source Ipv4 range field. Allow all ports.
+
+### 3. Edit Cloud Run settings
+
+As the data store is private, you need to configure your cloud run service's network setting to the
+VPC network you just created.
+
+1. Go to the [Cloud Run](https://console.cloud.google.com/run). Select your service.
+2. Navigate to the **Networking** tab.
+3. Select **Internal Ingress**. Save changes.
+4. Once deployed, click **Edit & deploy new version**.
+5. Edit container port to your datastore's port. Update `DFADDR` and `DFPASS` (empty if passkey is not set).
+6. Go to **Networking**. Select **Connect to a VPC for outbound traffic**. Choose your vpc.
```

Comment:
this is probably the most important part but it is buried in a lot of details about setting up the cloud run function.

I think we should start with some minimal clear steps that assumes you have a coloud run service and a private endpoint data store

Then all that is needed is:
1. Make sure dragonfly vpc is peer connected to the vpc cloud run is deployed too
2. *The *Connect to a VPC for outbound traffic**. Choose your vpc.setp
?

You can have a collapsable "end to end" example below if you want (or do it in a separate blog post which we can link form here)

Reply:
ok, sure makes sense.